src/scanner.c

```diff
@@ -269,6 +285,9 @@ ts_scanner_start_scan(ScannerCtx *ctx)
 			ctx->index = RelationGetRelid(ctx->indexrel);
 	}
 
+	Assert(ctx->internal.scan_mcxt != NULL);
```

Comment:
Related to the comment above, this seems to be an invariant, so maybe you should replace the `if` above with an `Assert`?

Reply:
Since we always set it if not set, this assert conveys this fact and assures that we didn't miss this somewhere.